### ISSUE TITLE
blocks %direction% %location% includes the original location and causes a cut off

### ISSUE THREAD

**Issue by Fusezion** [SkriptLang/Skript#6260]
### Skript/Server Version

```markdown
[03:31:27 INFO]: [Skript] Skript's aliases can be found here: https://github.com/SkriptLang/skript-aliases
[03:31:27 INFO]: [Skript] Skript's documentation can be found here: https://docs.skriptlang.org/
[03:31:27 INFO]: [Skript] Skript's tutorials can be found here: https://docs.skriptlang.org/tutorials
[03:31:27 INFO]: [Skript] Server Version: git-Paper-346 (MC: 1.20.4)
[03:31:27 INFO]: [Skript] Skript Version: 2.8.0-dev (selfbuilt-unknown)
[03:31:27 INFO]: [Skript] Installed Skript Addons:
[03:31:27 INFO]: [Skript]  - skript-gui v1.3 (https://github.com/APickledWalrus/skript-gui)
[03:31:27 INFO]: [Skript]  - skript-reflect v2.4-beta1 (https://github.com/SkriptLang/skript-reflect)
[03:31:27 INFO]: [Skript]  - SkBee v3.1.0 (https://github.com/ShaneBeee/SkBee)
[03:31:27 INFO]: [Skript] Installed dependencies: None
```

### Bug Description

Attempting to get any number of blocks from a direction using `blocks x %direction% of %location%` will only do X-1 blocks and then add the original block at defined location.

### Expected Behavior

Be consistent with the normal every day `block %direction% %location%` syntax i.e. `block 1 infront of target block` and `block east of target block`

### Steps to Reproduce

(view below images for repeatable syntax)

### Errors or Screenshots

Looooook! it's bad!!! Very inconsistent
![image](https://github.com/SkriptLang/Skript/assets/81952476/71f9b81d-a959-4520-9a65-5689a81b14d0)
![image](https://github.com/SkriptLang/Skript/assets/81952476/540720cc-072c-416b-84d6-39a9a05b4992)

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this report.

**Comment by sovdeeth** [comment]
This seems to only be the case with some directions:
a) in front acts like you described
![image](https://github.com/SkriptLang/Skript/assets/10354869/537fca07-a602-470b-9244-b7ec179fc8c2)
b) North works fine
![image](https://github.com/SkriptLang/Skript/assets/10354869/9e069cbc-f9bf-43cd-85bf-cebce439f90b)
c) South does not
![image](https://github.com/SkriptLang/Skript/assets/10354869/9d9d068d-6b1d-4092-8fff-2a1d30f92dbc)
b) Behind seems VERY strange but in a totally different way
![image](https://github.com/SkriptLang/Skript/assets/10354869/d06d5e14-6e96-475c-9ffb-4bf325dded44)

I think this may be due to the directions offsetting from a block's corner, rather than its center, so small floating point errors cause the output to change based on direction. I have not confirmed this theory at all.

**Comment by Fusezion** [comment]
The issue is likely the way the blocks with no facing work

I.e. defaults front to south

**Comment by sovdeeth** [comment]
I mean, yes, it's because front is South, but that's just a coincidence
The real issue is because of how the iterator works:
Ok after looking through the debugger as it did the iteration, it seems like the block iterator check the distance from the center of the block (block's NW corner + 0.5, 0.5, 0.5) to the original location.
When travelling North, West, or Up, this is fine because we get ~1.5 distance for the block 2 blocks away. Going South, East, and Down results in the distance being 2.5 for the block 2 blocks in that direction.

We could remove this "use the center of the block" correction, but I'm not sure what knockon effects this would have. We'd need to do some testing to make sure that's an appropriate solution:
![image](https://github.com/SkriptLang/Skript/assets/10354869/a7eea2a7-5bf0-4ab9-9410-c7db30b48b28)

**Comment by sovdeeth** [comment]
When using a centered location as the start of the iterator (this is what you tested, Fusezion), the behavior becomes consistent for all directions, but since the checker is a >**=** check, it stops on the last block, causing one less than expected:
![image](https://github.com/SkriptLang/Skript/assets/10354869/45b0e6e5-e8b6-498f-98d2-22dff1731f02)
![image](https://github.com/SkriptLang/Skript/assets/10354869/4f43eba1-13d7-4d99-8057-645c5ea994ad)

**Comment by Fusezion** [comment]
So basically just remove = and consistency becomes good. I personally am alright with this change probably will need to get opinions on the team

**Comment by sovdeeth** [comment]
The weird discrepancy with `block 2 behind {a}` was indeed to this line having floating point error that causes a slight movement to the side:
![image](https://github.com/SkriptLang/Skript/assets/10354869/4f949d5d-350b-494e-89cf-7dac20b13b71)
![image](https://github.com/SkriptLang/Skript/assets/10354869/3e6647cc-d2cd-46d9-a1bb-acf8da866c16)

**Comment by sovdeeth** [comment]
> So basically just remove = and consistency becomes good. I personally am alright with this change probably will need to get opinions on the team

Yes, I think that'd be ok for centered locations. I don't know if the behavior for non-centered locations should be using the distance to the block's center, but that's a different discussion.

**Comment by Fusezion** [comment]
putting note here just so I don't forget, this can be broken using `below` at a y level of `x.00`

![image](https://github.com/SkriptLang/Skript/assets/81952476/17d4a15e-7349-45a0-b150-069872af89c2)

**Comment by sovdeeth** [comment]
> putting note here just so I don't forget, this can be broken using `below` at a y level of `x.00`

This is the same thing I was talking about here:
> Ok after looking through the debugger as it did the iteration, it seems like the block iterator check the distance from the center of the block (block's NW corner + 0.5, 0.5, 0.5) to the original location.
When travelling North, West, or Up, this is fine because we get ~1.5 distance for the block 2 blocks away. Going South, East, and Down results in the distance being 2.5 for the block 2 blocks in that direction.

But I was a bit reductive, which directions will work as expected depends on exactly what position you're starting from.
